local ServerScriptService = game:GetService("ServerScriptService")
local Players = game:GetService("Players")

local EconomySystem = require(ServerScriptService.Server.EconomySystem)

Players.PlayerAdded:Connect(function(player)
	player.CharacterAdded:Wait()
	print(EconomySystem.GetBalance(player))
	print('giving 100')
	EconomySystem.GiveMoney(player)
	task.wait(2)
	print('giving 100')
	EconomySystem.GiveMoney(player, 100)
	task.wait(2)
	print('nvm taking it back')
	EconomySystem.RemoveMoney(player, 100)
end)